.github/workflows/ci-linux.yml

```diff
@@ -92,29 +137,74 @@ jobs:
     runs-on: ubuntu-22.04
     steps:
     - uses: actions/checkout@v2
-    - uses: ./.github/actions/install-essential-dependences
+    - uses: ./.github/actions/install-essential-dependencies
     - uses: ./.github/actions/init-make-config
       with:
         options: --cc=clang --cxx=clang++ --werror
     - name: compile
       run: |
-           make -j ${{env.proc_num}}
+        make -j ${{env.proc_num}}
+
+  clang-compile-with-make-protobuf-3-5-1:
```

Comment:
There are too many combinations. I seems that Apache had informed that no more than 20 jobs in one CI process. Maybe we can simplify these combinations. When we change one parameter (such as protobuf version), we keep the other parameters fixed (such as compiler type, build tool, OS version).

Reply:
I have merged some jobs and now there are only 15 jobs.